infra/metadata_harmonizer_step_functions/lambdas/harmonize_metadata.py

```
@@ -0,0 +1,874 @@
+            s3.put_object(
+                Bucket=batch_bucket,
+                Key=results_key,
+                Body=json.dumps(results_data, indent=2, default=str).encode(
+                    "utf-8"
+                ),
+                ContentType="application/json",
+            )
+
+            results_path = f"s3://{batch_bucket}/{results_key}"
+            logger.info(f"Results uploaded to {results_path}")
+        except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)